test/extended/util/framework.go

```diff
@@ -1691,15 +1691,8 @@ type podExecutor struct {
 
 // NewPodExecutor returns an executor capable of running commands in a Pod.
 func NewPodExecutor(oc *CLI, name, image string) (*podExecutor, error) {
-	out, err := oc.Run("run").Args(name, "--labels", "name="+name, "--image", image, "--restart", "Never", "--command", "--", "/bin/bash", "-c", "sleep infinity").Output()
-	if err != nil {
-		return nil, fmt.Errorf("error: %v\n(%s)", err, out)
-	}
-	_, err = WaitForPods(oc.KubeClient().CoreV1().Pods(oc.Namespace()), ParseLabelsOrDie("name="+name), CheckPodIsReady, 1, 3*time.Minute)
-	if err != nil {
-		return nil, err
-	}
-	return &podExecutor{client: oc, podName: name}, nil
+	execPod := CreateExecPodOrFail(oc.AdminKubeClient(), oc.Namespace(), name)
```

Comment:
The is probably the simplest solution, but it doesn't
1. Exit early on failure
2. Or dump logs or termination message

Is that okay? Is there any reason my DNS E2E function should do things more robustly than all of the other code that calls `CreateExecPodOrFail`?